src/python/zquantum/core/circuit/conversions/pyquil_conversions.py

```diff
@@ -50,18 +52,32 @@
 TWO_QUBIT_CONTROLLED_GATES = {
     CZ: pyquil.gates.CZ,
     CNOT: pyquil.gates.CNOT,
-    SWAP: pyquil.gates.SWAP,
 }
 
 
+PYQUIL_NAME_TO_ORQUESTRA_CLS = {
+    cls.__name__: cls
+    for cls in chain(
+        SINGLE_QUBIT_NONPARAMETRIC_GATES, TWO_QUBIT_CONTROLLED_GATES, ROTATION_GATES
+    )
+}
+
+PYQUIL_NAME_TO_ORQUESTRA_CLS["CPHASE"] = CPHASE
+PYQUIL_NAME_TO_ORQUESTRA_CLS["SWAP"] = SWAP
```

Comment:
These seem like outliers and I think it's fine for now, but it's probably unsustainable in the long-term to have these act as "special cases". Maybe, for the future, we can think about better ways to describe/categorize gates so that we run into these cases less frequently.

Reply:
Also, I feel like saying `CPHASE` isn't a "controlled two qubit gate" isn't true. Maybe best to rename `TWO_QUBIT_CONTROLLED_GATES` to `TWO_QUBIT_NONPARAMETRIC_CONTROLLED_GATES`?

Reply:
A possible longer-term solution would be to have a map from gate name to a tuple of `(number_of_qubits: int, is_parametric: bool, is_controlled: bool)` and then build logic off of that knowledge?

Reply:
I like the idea of building logic based on the map describing properties of gates, but see comment below about the SWAP gate. Also, you are right that `TWO_QUBIT_NONPARAMETRIC_CONTROLLED_GATES` is a better name, I changed it in 8781468 and in 2a7a4e8 I added comment about the purpose of  `PYQUIL_NAME_TO_ORQUESTRA_CLS`.